independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ObserverGenerator.java

```diff
@@ -226,6 +226,7 @@ Collection<Resource> generate(ObserverInfo observer) {
 
         implementGetBeanClass(observerCreator, observer.getBeanClass());
         implementNotify(observer, observerCreator, injectionPointToProviderField, reflectionRegistration, isApplicationClass);
+        implementGetReception(observerCreator, observer);
```

Comment:
We could implement this method only for observers with `Reception.IF_EXISTS` and use the default method for the rest (99% of cases). We use this strategy for other parts of the API in order to generate smaller classes.

Reply:
That was in fact my initial impl but the difference is so minimal that I thought it insignificant. But I have nothing against that, so let me revert to default impl + conditional code generation.

Reply:
Done